src/Adapter/ReactPhp/EventLoop.php

```diff
@@ -86,18 +86,12 @@ function ($reason) use ($generator, $fnSuccess, $fnFailure, $fnWrapGenerator) {
             }
         };
 
-        $deferred = new Internal\Deferred();
+        $deferred = Internal\Deferred::forAsync();
         $fnWrapGenerator(
             $generator,
             [$deferred, 'resolve'],
             function (\Throwable $throwable) use ($deferred) {
-                if ($deferred->getPromiseWrapper()->hasBeenYielded()) {
-                    $deferred->reject($throwable);
-                } else {
-                    $this->reactEventLoop->futureTick(function () use ($throwable) {
-                        throw $throwable;
-                    });
-                }
+                $deferred->reject($throwable);
```

Comment:
You could:
* use `[$deferred, 'reject']`, in the same way than the line above
* directly inject `$deferred` in `$fnWrapGenerator`, like you did for `Amp`

I think the second option is clearer, so I prefer it 😃